src/pilot/config_idrac.py

```diff
@@ -415,33 +427,48 @@ def config_idrac(instack_lock,
         ip_service_tag,
         node_definition)
 
-    # Configure the NIC port to PXE boot or not
-    reboot_required_nic, nic_job_ids, provisioning_mac = \
-        configure_nics_boot_settings(drac_client,
-                                     ip_service_tag,
-                                     pxe_nic_fqdd,
-                                     node,
-                                     target_boot_mode)
-
-    reboot_required = reboot_required or reboot_required_nic
-    if nic_job_ids:
-        job_ids.extend(nic_job_ids)
-
-    # Do initial idrac configuration
-    reboot_required_idrac, idrac_job_id = config_idrac_settings(drac_client,
-                                                                ip_service_tag,
-                                                                password,
-                                                                node)
-    reboot_required = reboot_required or reboot_required_idrac
-    if idrac_job_id:
-        job_ids.append(idrac_job_id)
-
-    # If we need to reboot, then add a job for it
-    if reboot_required:
-        LOG.info("Rebooting {} to apply configuration".format(ip_service_tag))
-
-        job_id = drac_client.create_reboot_job()
-        job_ids.append(job_id)
+    boot_modes_same = BootModeHelper.get_boot_mode(
+        drac_client) == target_boot_mode
+
+    if skip_nic_config and boot_modes_same:
+        provisioning_mac = skipped_configure_nics_boot_settings(
+            drac_client, pxe_nic_fqdd)
+        LOG.info("Skipping NIC configuration")
+
+    else:
+
+        if skip_nic_config and not boot_modes_same:
+            LOG.warning("NIC configurations cannot be skipped because"
+                        "current and target boot modes are different")
+        # Configure the NIC port to PXE boot or not
+        reboot_required_nic, nic_job_ids, provisioning_mac = \
+            configure_nics_boot_settings(drac_client,
+                                         ip_service_tag,
+                                         pxe_nic_fqdd,
+                                         node,
+                                         target_boot_mode)
+
+        reboot_required = reboot_required or reboot_required_nic
+        if nic_job_ids:
+            job_ids.extend(nic_job_ids)
+
+        # Do initial idrac configuration
```

Comment:
This section of code that deals with idrac configuration needs to be executed even if the user chooses to skip NIC configuration.  This is because the user may choose to change the password on the iDRAC, and that will only happen when config_idrac_settings is called.  I think you can just un-indent 455-471 to achieve this.

Reply:
done